src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -252,20 +253,30 @@ private IEnumerable<DiagnosticAnalyzer> GetAnalyzersForTypeNames(Assembly analyz
             // Given the type names, get the actual System.Type and try to create an instance of the type through reflection.
             foreach (var typeName in analyzerTypeNames)
             {
-                DiagnosticAnalyzer analyzer = null;
+                Type type;
                 try
                 {
-                    var type = analyzerAssembly.GetType(typeName);
-                    if (DerivesFromDiagnosticAnalyzer(type))
-                    {
-                        analyzer = (DiagnosticAnalyzer)Activator.CreateInstance(type);
-                    }
+                    type = Type.GetType(typeName + ", " + analyzerAssembly.FullName, throwOnError: true);
+                }
+                catch (Exception e)
+                {
+                    this.AnalyzerLoadFailed?.Invoke(this, new AnalyzerLoadFailureEventArgs(AnalyzerLoadFailureEventArgs.FailureErrorCode.UnableToCreateAnalyzer, e.Message, e, typeName));
+                    reportedError = true;
+                    continue;
+                }
+
+                Debug.Assert(type != null);
+
+                DiagnosticAnalyzer analyzer;
+                try
+                {
+                    analyzer = Activator.CreateInstance(type) as DiagnosticAnalyzer;
```

Comment:
Why `as` here instead of a direct cast?  It seems like the current form won't issue an error message when there is a type conversion error.